filters/replicateto/replicateto.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2015 Zalando SE
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package replicateto
+
+import (
+	"fmt"
+	"github.com/zalando/skipper/filters"
+	"net/http"
+	"net/url"
+	"regexp"
+)
+
+const (
+	Name = "replicateTo"
+)
+
+type replicateToSpec struct{}
+
+type replicateType int
+
+const (
+	asBackend replicateType = iota + 1
+	pathModified
+)
+
+type replicateTo struct {
+	typ         replicateType
+	host        string
+	scheme      string
+	rx          *regexp.Regexp
+	replacement []byte
+}
+
+// Returns a new replication filter Spec, whose instances execute
+// exactly same Request against a shadow backend.
+// parameters: shadow backend url, optional - the expression to match and the replacement string.
+// Name: "replicateTo".
+
+// Example
+// Path("http://example.com/api/v3") -> replicateTo("https://api.example.com") -> "http://example.org/"
+//This route wil send incoming query to http://example.org/api/v3 but will also send
+// a copy of the query to https://api.example.com/api/v3. Note that scheme is changed
+
+// Example
+// Path("http://example.com/api/v3") -> replicateTo("https://api.example.com", ".*", "/v1/" ) -> "http://example.org/"
+//This route wil send incoming query to http://example.org/api/v3 but will also send
+// a copy of the query to https://api.example.com/v1/ . Note that scheme and path are changed
+
```

Comment:
This empty line renders the doc invisible in the generated doc, pls remove the empty line here. You can test the docs with `godoc -http :8080`.